### PR TITLE
[DX-1207] Add Core SDK v1.0.0-beta.1 version to docs

### PR DESCRIPTION
**Pull request by matthewmuscat**
## Description
This PR:
- Adds autogenerated SDK reference
- Adds new entry for v1.0.0-beta.1 to the version switcher
- Fixes styling on SDK version switcher

## Resolved issues
Jira: https://immutable.atlassian.net/browse/DX-1207

### Before submitting the PR, please consider the following:
- [ ] It's beneficial if your pull request references an issue used to discuss the pull request ahead of time. If you haven't previously created an issue, please create one and discuss your contribution with the maintainers.
- [ ] Prefix your PR title with `feat: `, `fix: `, `chore: `, `docs:`, or `refactor:`.
- [ ] The description should clearly illustrate what problems the pull request solves.
- [ ] Ensure that the commit messages follow our guidelines.
- [ ] Resolve merge conflicts (if any).
- [ ] Make sure that the current branch is up to date with the `main` branch.

### For internal Immutable developers:
- [ ] If you are adding or updating a documentation page, please also add or update the team ownership of that page (follow [this guide](https://immutable.atlassian.net/wiki/spaces/PPS/pages/1947927045/Docs+ownership+registry#What-to-do-when-adding%2Fupdating-documentation-pages%3F)).
- [ ] If you are adding or updating documentation for an SDK, please make sure that you meet the requirements in [this doc](https://immutable.atlassian.net/wiki/spaces/PPS/pages/1916994017/SDK+documentation+guide).
- [ ] Please obtain PR approval from all of the following:
    - Your tech lead (Fallback: Team lead or engineering manager)
    - Your product manager (Fallback: Group PM)
    - Another software engineer on your team

### CODE REVIEW

File: sdks-core-sdk-ts_versioned_sidebars/version-1.0.0-beta.1-sidebars.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+      "label": "Core SDK TypeScript",
+      "items": [
+        "overview",
+        "installation",
+        "initialization",
+        "additional-info",
+        {
+          "type": "link",
+          "label": "Reference",
+          "href": "https://docs.x.immutable.com/sdk-references/core-sdk-ts/1.0.0-beta.1/classes/ImmutableX.html"
```
Comment (zatlite) [inline sdks-core-sdk-ts_versioned_sidebars/version-1.0.0-beta.1-sidebars.json:14]: should it be ImmutableX.ImmutableX.html like the other?

File: docs/sdk-docs/core-sdk-ts/overview.md [hunk 2]
```diff
@@ -8,7 +8,7 @@ keywords: [imx-dx]
 
 # Overview
 
-:::danger The Core TypeScript SDK Interface is under active development and hasn't hit v1.0 yet.
+:::caution The Core TypeScript SDK Interface is currently in beta and is expected to in a stable v1.0 release in the near short term.
```
Comment (Kastet) [inline docs/sdk-docs/core-sdk-ts/overview.md:11]:
```suggestion
:::caution The Core TypeScript SDK Interface is currently in beta and is expected to be a stable v1.0 release in the near short term.
```

**Review by CodeSchwert — APPROVED** [review]

**Comment by CodeSchwert** [comment]
We should stop mixing auto generated SDK reference files in with other files. Maybe make a seperate PR for that. It make reviewing changes a bit of a pain.

**Comment by matthewmuscat** [comment]
> We should stop mixing auto generated SDK reference files in with other files. Maybe make a seperate PR for that. It makes reviewing changes a bit of a pain.

Apologies 👍 they needed to be here for it all to work. If you review via commits it makes it much easier as I specified what was autogen and what wasnt.

**Review by Kastet — APPROVED** [review]

**Pull request merged**